### ISSUE TITLE
[BUG] Trace has status "In Progress" indefinitely when using autolog with llamaindex Workflow

### ISSUE THREAD

**Issue by tordks** [mlflow/mlflow#13352]
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Local machine

### Willingness to contribute

No. I cannot contribute a bug fix at this time.

### MLflow version

- Client: 2.16.2
- Tracking server: 2.16.2

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 22.04.2
- **Python version**: 3.12.5

### Describe the problem

Hi,

I am having an issue with tracing a llamaindex workflow using `mlfow.llama_index.autolog`.  The code example executes a workflow with a single step that calls `llm.acomplete` and prints the result. mlflow registeres that a trace is started. However, the trace is never completed and has status "In progress" within the UI indefinitely, as seen in the image below. Expected behavior would be that the trace finished and showed status OK.

![Screenshot 2024-10-08 130517](https://github.com/user-attachments/assets/c5bdd93f-fd6a-4ec3-9d0a-1e584fcdf00b)

If I remove the Workflow wrapper the trace completes as expected:

![Screenshot 2024-10-08 131020](https://github.com/user-attachments/assets/5361cd32-696b-482d-bab0-a80ac5157e1a)

### Tracking information

```shell
MLflow version: 2.16.2
Tracking URI: xxx
Artifact URI: xxx
System information: Linux #1 SMP Fri Mar 29 23:14:13 UTC 2024
Python version: 3.12.5
MLflow version: 2.16.2
MLflow module location: xxx
Tracking URI: xxx
Registry URI: xxx
Active experiment ID: 0
Active run ID: 09cdb040b97d4ba491db603e906f6eff
Active run artifact URI: xxx
MLflow dependencies: 
  Flask: 3.0.3
  Jinja2: 3.1.4
  aiohttp: 3.10.5
  alembic: 1.13.3
  docker: 7.1.0
  graphene: 3.3
  gunicorn: 23.0.0
  markdown: 3.7
  matplotlib: 3.9.2
  mlflow-skinny: 2.16.2
  numpy: 1.26.4
  pandas: 2.2.3
  pyarrow: 17.0.0
  pydantic: 2.9.2
  scikit-learn: 1.5.2
  scipy: 1.14.1
  sqlalchemy: 2.0.35
  tiktoken: 0.7.0
```

### Code to reproduce issue

Example where the trace has status "In Progress" indefinitely:

```python
import asyncio
import mlflow
from llama_index.core.workflow import (
    StartEvent,
    StopEvent,
    Workflow,
    step,
)

from llama_index.llms.openai import OpenAI


class JokeFlow(Workflow):
    llm = OpenAI()

    @step
    async def generate_joke(self, ev: StartEvent) -> StopEvent:
        topic = ev.topic

        prompt = f"Write your best joke about {topic}."
        response = await self.llm.acomplete(prompt)
        return StopEvent(result=str(response))

async def main():
    w = JokeFlow(timeout=60, verbose=False)
    result = await w.run(topic="pirates")
    print(str(result))

if __name__ == "__main__":
    mlflow.set_experiment("minimal-flow")
    mlflow.llama_index.autolog()

    asyncio.run(main())
```

Same code, except the Workflow, where the trace gets the expected "OK" status:

```python
import asyncio
import mlflow

from llama_index.llms.openai import OpenAI

async def main():
    llm = OpenAI()

    topic = "pirates"

    prompt = f"Write your best joke about {topic}."
    joke = await llm.acomplete(prompt)
    print(joke)


if __name__ == "__main__":
    mlflow.set_experiment("minimal-flow-no-workflow")
    mlflow.llama_index.autolog()

    asyncio.run(main())
```

### Stack trace

stack trace not relevant.

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by harupy** [comment]
@tordks Thanks for reporting this. I ran the code but could not reproduce the issue. Not sure if this is related, but what backend did you use?

**Comment by B-Step62** [comment]
@tordks This will be resolved by 2.17.0 release coming soon, combined with a [fix](https://github.com/run-llama/llama_index/pull/16290) on LlamaIndex side released in 0.11.16!

**Comment by tordks** [comment]
@harupy Apologies for not responding earlier. I was using the default file system backend.

@B-Step62 That is awesome, thanks you!

Can confirm that I now get a trace by updating mlflow and llamaindex.
